app/etc/di.xml

```diff
@@ -1756,4 +1756,61 @@
             </argument>
         </arguments>
     </type>
+    <preference for="\Magento\Framework\GraphQl\Query\ErrorHandlerInterface" type="\Magento\Framework\GraphQl\Query\ErrorHandler"/>
+    <type name="\Magento\Framework\GraphQl\Query\ErrorHandler">
+        <arguments>
+            <argument name="clientLogger" xsi:type="object">GraphQLClientLogger</argument>
+            <argument name="serverLogger" xsi:type="object">GraphQLServerLogger</argument>
+            <argument name="generalLogger" xsi:type="object">GraphQLGeneralLogger</argument>
+            <argument name="clientErrorCategories" xsi:type="array">
+                <item name="graphql" xsi:type="const">\GraphQL\Error\Error::CATEGORY_GRAPHQL</item>
+                <item name="alreadyExists" xsi:type="const">\Magento\Framework\GraphQl\Exception\GraphQlAlreadyExistsException::EXCEPTION_CATEGORY</item>
+                <item name="authentication" xsi:type="const">\Magento\Framework\GraphQl\Exception\GraphQlAuthenticationException::EXCEPTION_CATEGORY</item>
+                <item name="authorization" xsi:type="const">\Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException::EXCEPTION_CATEGORY</item>
+                <item name="input" xsi:type="const">\Magento\Framework\GraphQl\Exception\GraphQlInputException::EXCEPTION_CATEGORY</item>
+                <item name="noSuchEntity" xsi:type="const">\Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException::EXCEPTION_CATEGORY</item>
+                <item name="request" xsi:type="string">request</item>
```

Comment:
Looks like it's lack of GraphQL exception classes

Reply:
I can suggest 2 ways to divide errors into client and server:

- Use \GraphQL\Error\ClientAware::isClientSafe
- Just check if the exception is an instance of \Magento\Framework\Exception\LocalizedException

Reply:
@naydav - in case we will use `\GraphQL\Error\ClientAware::isClientSafe` for logger detection there will be no need to put all of these classes here.